### DISCUSSION TITLE
Extension of C# Client Generator

### DISCUSSION THREAD

**Discussion by ImoutoChan** [RicoSuter/NSwag#3893] [Q&A] [answered]
Sometimes I have a need to generate additional methods for generated clients. I would like to know if there is an ability to extend generating functionality with my own logic?

Why do I want this? Because I want to generate additional GET methods that would handle 404 errors and return default values in such cases. Something like GetOrDefault(...). 

If there is no way to extend existing functionality, maybe you can advise a way how I can approach my problem from a different angle.

**Comment by RicoSuter** [comment, marked as answer]
>Sometimes I have a need to generate additional methods for generated clients. I would like to know if there is an ability to extend generating functionality with my own logic?

All generated C# classes are partial, so you can easily extend it with additional methods.

**Reply by ImoutoChan** [reply]
I wrote a source generator for generated c# classes. It works perfectly, thank you!
